firebase_functions/functions/src/migration.ts

```diff
@@ -0,0 +1,150 @@
+import { Client, ClientBuilder } from '@iota/client';
+import * as admin from 'firebase-admin';
+import * as crypto from 'crypto';
+
+export const migration = async (from: number, to: number) => {
+  await deleteUnusedDevices();
+  await deleteUnusedUsers(from, to);
+  await updateDevices(from, to);
+  await updateUser(from, to);
+};
+
+const deleteUnusedDevices = async () => {
+  const deviceRef = admin.firestore().collection('devices');
+  await deviceRef.get().then(async (querySnapshot) => {
+    const docs = querySnapshot.docs;
+    console.log("Docs length: ", docs.length)
+    for (let docIndex = 50; docIndex < 174; docIndex++ ) {
```

Comment:
From where do you get these numbers, 50 and 174?

Reply:
Fixed, replaced the hard coded numbers with vars